pkg/endpoint/endpoint.go

```diff
@@ -1171,6 +1171,14 @@ func (e *Endpoint) SetK8sMetadata(containerPorts []slim_corev1.ContainerPort) er
 	return nil
 }
 
+// GetK8sPorts returns the k8sPorts, which must not be modified by the caller
+func (e *Endpoint) GetK8sPorts() (k8sPorts policy.NamedPortMap) {
+	e.mutex.RLock()
```

Comment:
I'm not sure I understand the locking here. `pkg/endpoint/lock.go` has lock wrappers that attempt to protect us from various cases by for example propagating back up an error in the case that the endpoint disappears. 

We even have the `unconditionalRLock()` variations, including documentation, which do exactly what this is doing but have conditional comments to describe cases where it's OK to do this vs. `rlockAlive()`. Surely we should be using the wrappers rather than directly interacting with `e.mutex`, even if we decide that the actual logic here is OK because the caller has a reference to the endpoint object?

Reply:
Thanks for the reminder, I had completely overlooked this. Fixed.